src/librustc_mir/dataflow/move_paths/builder.rs

```diff
@@ -151,11 +153,13 @@ impl<'b, 'a, 'gcx, 'tcx> Gatherer<'b, 'a, 'gcx, 'tcx> {
                             place.clone()
```

Comment:
This `place` is wrong. It should be the one at the current projection level, but it's always the outermost one. I think you can reconstruct the place by `Place::Projection(PlaceProjection { base: proj.base, elem: proj.elem.clone() })`